### DISCUSSION TITLE
Save a planned route in favorites

### DISCUSSION THREAD

**Discussion by GuG1050** [devemux86/cruiser#472] [Q&A] [answered]
May be it is too obvious .... but I can not find how to save a cruiser-planned route directly in My Placees -> Routes. 
My workaround is  to export and then to import my Route into My Places -> Routes. Is there an other way?

**Comment by devemux86** [comment, marked as answer]
@GuG1050 

Long press on the route and select "Save favorite route".
(2nd screen)

<img src="https://github.com/devemux86/cruiser/assets/3484020/766d2754-d666-4158-a429-a3301c1e2296" width="50%"/>

**Reply by GuG1050** [reply]
👍 thank you.

**Reply by bios4** [reply]
Maybe you could make it a little bit more clear/intuitive @devemux86:

- Show "Save **_as_** favorite route" when it is a new route and not yet stored in the collection
- Show "**_Update_** favorite route" when the currently loaded route is already a route in "My places"

**Reply by devemux86** [reply]
@bios4 

If we change the filled name, then it is not updated, but is saved as a new route.
So the save depends on what we want to do.

**Reply by bios4** [reply]
What do you mean with "filled name"?
Do you mean that there are three use cases to be considered???

1) Route IS NOT yet in my collection - "**_Save as_** favorite route"
2) Route IS in my collection, but was updated - show "**_Update_** favorite route"
2) Route IS in my collection, was updated and should be stored as a new route - show "**_Save as new_** favorite route"

**Reply by devemux86** [reply]
"Filled name" is the route name in the save dialog.

Changing the menu text to "update" means that it can only update an existing route, not save as a new route.
We need both options available, save as new route or update the existing route.

Now we check the name, if it is the same, we update, if it is different, we save as new route.
We probably need a better workflow, while keeping all options available.

**Reply by bios4** [reply]
> Now we check the name, if it is the same, we update, if it is different, we save as new route.

Could you apply exactly this check also to determine which translation string to be displayed??? 🤔
Or do you mean that this information is only available AFTER the user clicked the tile?

**Reply by devemux86** [reply]
The information is available after the save dialog, because the save dialog holds the final selected route name.

**Reply by bios4** [reply]
I see...

**Comment by devemux86** [comment]
A simpler workflow is to save / overwrite the planned route in the favorite routes screen (instead of the map):
- In favorite route groups use "Save planned route" to save as new route in the group (image 1)

- In favorite routes use "Overwrite with planned route" to update the selected route (image 2)

<img src="https://github.com/devemux86/cruiser/assets/3484020/4cfe807f-4648-4e42-9230-bc3fc29e82bd" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/4e974bcd-8069-4a82-b76c-fdb56a847c04" width="50%"/>

**Reply by devemux86** [reply]
> In favorite route groups use "Save planned route" to save as new route in the group
> In favorite routes use "Overwrite with planned route" to update the selected route

The new way to save or overwrite the planned route in favorites is available for testing in:
- https://github.com/devemux86/cruiser/discussions/482
